File: backend/app/schemas/transaction.py
from sqlmodel import SQLModel
from datetime import datetime
from app.models.transaction import TransactionType

class TransactionRead(SQLModel):
    id: int
    description: str
    amount: float
    date: datetime
    type: TransactionType
    category_id: int | None
    created_at: datetime
    updated_at: datetime

class TransactionCreate(SQLModel):
    description: str
    amount: float
    date: datetime
    type: TransactionType
    category_id: int | None

class TransactionUpdate(SQLModel):
    description: str | None = None
    amount: float | None = None
    date: datetime | None = None
    type: TransactionType | None = None
    category_id: int | None = None

File: frontend/src/types/transaction.ts
export type TransactionType = "income" | "expense";

export interface Transaction {
  id: number;
  description: string;
  amount: number;
  date: string;
  type: TransactionType;
  category_id: number;
  created_at: string;
  updated_at: string;
}

export interface TransactionForm {
  description: string;
  amount: number;
  date: string;
  type: TransactionType;
  category_id: number | null;
}

export type Category = {
  id: number;
  name: string;
  type: TransactionType;
};